Payjoin-spec.md

```diff
@@ -0,0 +1,182 @@
+
+
+\* <a name="unsupported-inputs"></a>: We currently only support receivers and senders using P2WPKH, while we plan to relax this in a later release, it is important for the receiver to use the same input's type as the sender, otherwise the payjoin transaction would provide no privacy benefit. Since most wallet do not support P2WSH or wrapped P2SH for the sender, we decided to not offer this feature for now.
+
+\*\* <a name="poisoning"></a>: If the sender overpays the invoice, we can substract fees from his over paid amount. This is especially useful when somebody wants to make a donation from a small unspent output in his own wallet.
+
+### Sender side
+
+Before sending submitting the original PSBT to the receiver, the sender make sure that the original PSBT is broadcasted whatever happen by automatically broadcasting it after 1 minute.
+
+The sender then follow the steps:
+
+* Check that the PSBT must not contain global xpub, hd key path or public key information.
+* Check that the PSBT is sane.
+* Check that all the spent outpoints in the original PSBT still exists in the coinjoin PSBT.
+* Check that all the spent outpoints in the original PSBT does not have any partial signature.
+* Check that the receiver added at least one input.
+* Check that the receiver added P2WPKH inputs.
+* Check that any other outpoints (those added by the receiver) are finalized.
+* Check that the transaction version, and nLockTime are unchanged.
+* Check that the sender's input's sequence numbers are unchanged.
+* Update `WitnessUTXO` of the PSBT.
+* Fill out only the HD Key paths and public keys of inputs and outputs that were present in the original PSBT.
+* Check that the sent amount in the payjoin transaction is the same as the sent amount of the original transaction. [*](#calculate-balance)
+* If those are different:
+    * Check that the additional paid amount is not more than the additional fee paid in the payjoin transaction.
+    * Check that the additional paid amount is not more than twice the original fee 
+    * Check that the additional paid amount is not more than the expected fee once the payjoin transaction is fully signed [**](#calculate-increased-fee)
+
+After all those checks, the sender can proceed to sign the payjoin transaction.
+
+Note: 
+
+* The sender **does NOT check** whether ouputs have been removed or modified. This allow flexibility to the receiver to adapt his receiving address type to match the other outputs's address type of the sender. For example, if the receiver support both P2WPKH and P2WSH, while the invoice's address in the original transaction was P2WPKH, the receiver may change the address to be P2WSH to match customer's change address format. This is safe because the sender only cares that he does not send too much money in the payjoin transaction. It is also useful if the receiver wants to batch some of his own payments in the transaction.
```

Comment:
```
The sender **does NOT check** whether ouputs have been removed or modified. This allows flexibility to the receiver to adapt his receiving address type to match the other outputs' address type of the sender. For example, if the receiver support both P2WPKH and P2WSH, while the invoice's address in the original transaction was P2WPKH, the receiver may change the address to be P2WSH to match customer's change address format. This is safe because the sender only cares that he does not send too much money in the payjoin transaction. It is also useful if the receiver wants to batch some of his own payments in the transaction.
```